File: uiframe/src/main/java/mobi/cangol/mobile/base/BaseActivityDelegate.java
/*
 * Copyright (c) 2013 Cangol
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License")
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package mobi.cangol.mobile.base;

import android.os.Bundle;
import android.os.Handler;
import androidx.annotation.AttrRes;
import androidx.annotation.ColorInt;
import android.widget.EditText;

import mobi.cangol.mobile.service.AppService;
import mobi.cangol.mobile.service.session.Session;

public interface BaseActivityDelegate {

    /**
     * 查找view
     */
    void findViews();

    /**
     * 初始化view
     *
     * @param savedInstanceState
     */

    void initViews(Bundle savedInstanceState);

    /**
     * 初始化数据
     *
     * @param savedInstanceState
     */
    void initData(Bundle savedInstanceState);

    /**
     * 显示toast
     *
     * @param resId
     */
    void showToast(int resId);

    /**
     * 显示toast
     *
     * @param str
     */
    void showToast(String str);

    /**
     * 显示toast
     *
     * @param resId
     * @param duration
     */
    void showToast(int resId, int duration);

    /**
     * 显示toast
     *
     * @param str
     * @param duration
     */
    void showToast(String str, int duration);

    /**
     * 获取AppService
     *
     * @param name
     * @return
     */
    AppService getAppService(String name);

    /**
     * 获取Session
     *
     * @return
     */
    Session getSession();

    /**
     * 设置全屏
     *
     * @param fullscreen
     */
    void setFullScreen(boolean fullscreen);
    /**
     * 是否是全屏
     *
     */
    boolean isFullScreen();
    /**
     * back按钮回调
     */
    void onBack();


    /**
     * 显示软键盘
     *
     * @param editText
     */
    void showSoftInput(EditText editText);

    /**
     * 隐藏软键盘
     *
     */
     void hideSoftInput();

    /**
     * 隐藏软键盘
     *
     */
    void hideSoftInput(EditText editText);

    /**
     * 获取主线程的Handler
     */
    Handler getUiHandler();

    /**
     * 获取共享线程的Handler
     */
    Handler getThreadHandler();

    @ColorInt
    int getThemeAttrColor(@AttrRes int colorAttr);

}

File: uiframe/src/main/kotlin/mobi/cangol/mobile/base/BaseActivityDelegate.kt
package mobi.cangol.mobile.base

import android.os.Bundle
import android.os.Handler
import android.support.annotation.AttrRes
import android.support.annotation.ColorInt
import android.widget.EditText
import mobi.cangol.mobile.service.AppService
import mobi.cangol.mobile.service.session.SessionService

interface BaseActivityDelegate {

    /**
     * 获取Session
     *
     * @return
     */
    fun getSession(): SessionService

    /**
     * 设置全屏
     *
     * @param fullscreen
     */
    fun setFullScreen(fullscreen: Boolean)

    /**
     * 是否全屏
     *
     */
    fun isFullScreen(): Boolean

    /**
     * 获取一个主线程的Handler
     */
    fun getHandler(): Handler

    /**
     * 查找view
     */
    fun findViews()

    /**
     * 初始化view
     *
     * @param savedInstanceState
     */

    fun initViews(savedInstanceState: Bundle?)

    /**
     * 初始化数据
     *
     * @param savedInstanceState
     */
    fun initData(savedInstanceState: Bundle?)

    /**
     * 显示toast
     *
     * @param resId
     */
    fun showToast(resId: Int)

    /**
     * 显示toast
     *
     * @param str
     */
    fun showToast(str: String)

    /**
     * 显示toast
     *
     * @param resId
     * @param duration
     */
    fun showToast(resId: Int, duration: Int)

    /**
     * 显示toast
     *
     * @param str
     * @param duration
     */
    fun showToast(str: String, duration: Int)

    /**
     * 获取AppService
     *
     * @param name
     * @return
     */
    fun getAppService(name: String): AppService

    /**
     * back按钮回调
     */
    fun onBack()


    /**
     * 显示软键盘
     *
     * @param editText
     */
    fun showSoftInput(editText: EditText)

    /**
     * 隐藏软键盘
     *
     */
    fun hideSoftInput()

    /**
     * 隐藏软键盘
     *
     */
    fun hideSoftInput(editText: EditText)


    @ColorInt
    fun getThemeAttrColor(@AttrRes colorAttr: Int): Int

}
